packages/sanity/src/structure/panes/document/documentPanel/header/DocumentPanelHeader.tsx

```diff
@@ -130,8 +135,34 @@ export const DocumentPanelHeader = memo(
                 <PaneHeaderActionButton key={item.key} node={item} />
               ))}
 
-              <PaneContextMenuButton nodes={contextMenuNodes} key="context-menu" />
-
+              <RenderActionCollectionState
+                actions={actions || []}
+                // @ts-expect-error TODO: fix the document actions
```

Comment:
If we're unable to remove this before merging, could this TODO comment elaborate a bit about what the fix is?

Reply:
Fixed, thanks! It required to update the ts definition of `GetHookCollectionState`.
This component is adding the `onComplete` action which is missing in the `editStateFor`, that's why the types weren't matching. 
https://github.com/sanity-io/sanity/pull/5933/commits/0fbdbb20c2f7f9a9507026b158a04fd00442088a#diff-d6b1504a2d940142d300df2888f280863ff5ae4c51762fed870301296ec72877